src/AssetManagement/actions/UniswapV2SwapAction.sol

```diff
@@ -0,0 +1,94 @@
+/**
+ * Created by Arcadia Finance
+ * https://www.arcadia.finance
+ *
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+pragma solidity >=0.8.0 <0.9.0;
+
+import "../ActionBase.sol";
+import "../helpers/UniswapV2Helper.sol";
+import "../utils/ActionAssetData.sol";
+import "../../interfaces/IMainRegistry.sol";
+import "../../interfaces/IVault.sol";
+import "../../interfaces/IERC20.sol";
+
+contract UniswapV2SwapAction is ActionBase, UniswapV2Helper {
+    constructor(address _router, address _mainreg) ActionBase(_mainreg) UniswapV2Helper(_router) {}
+
+    function executeAction(address _vaultAddress, bytes calldata _actionData)
+        public
+        override
+        returns (actionAssetsData memory)
+    {
+        require(_vaultAddress == msg.sender, "UV2_SWAP: can only be called by vault");
+        // preCheck data
+        (actionAssetsData memory _outgoing, actionAssetsData memory _incoming, address[] memory path) =
+            _preCheck(_actionData);
+        // execute Action
+        _execute(_outgoing, _incoming, path);
+        // postCheck data
+        _incoming.assetAmounts = _postCheck(_incoming);
```

Comment:
this shouldn't be set by the postcheck but should be the return data of swapExactTokensForTokens

Reply:
At this point I think I might aswel remove the entire postCheck since it it not doing anything anymore.